file/builder.go

```diff
@@ -250,6 +250,28 @@ func (b *stateBuilder) consumers() {
 			b.err = err
 			return
 		}
+
+		var mtlsAuths []kong.MTLSAuth
+		for _, cred := range c.MTLSAuths {
+			var username *string
+			var customID *string
+			if !utils.Empty(c.Username) {
+				username = kong.String(*c.Username)
+			}
+			if !utils.Empty(c.CustomID) {
+				customID = kong.String(*c.CustomID)
+			}
+			cred.Consumer = &kong.Consumer{
+				ID:       kong.String(*c.ID),
+				Username: username,
+				CustomID: customID,
+			}
```

Comment:
Yeah--essentially I'm just passing these to the cred builder/`ingestMTLSAuths` so it can complain in a useful manner, since we can't refer to mtls-auths by name (none exists) or ID (the errors appear if you omit it)--the best option I can think of to guide users to the broken object in that case is to present the named consumer it's tied to.

They get discarded after if they weren't needed for error messages; behavior does align with the others but it happens inside ingest at https://github.com/hbagdi/deck/blob/050b56c7b96bcbbab0e461a992af8028fffb079d/file/builder.go#L425-L432